net.go

```diff
@@ -116,13 +158,37 @@ func (l *Listener) Listen() error {
 		ll = ratelimit.NewListener(ll, wl, rl)
 	}
 
-	l.listener = ll
-	return nil
+	return ll, nil
+}
+
+func (l *Listener) acceptLoop(ll net.Listener) {
```

Comment:
I'd use `errgroup` with anonymous function instead of creating this and closeCh.